jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -129,25 +111,75 @@ private static native void checkpointJVMImpl(String imageDir,
 	 */
 	public CRIUSupport(Path imageDir) {
 		/* [IF JAVA_SPEC_VERSION < 17] */
+		@SuppressWarnings({"deprecation" })
+		/* [ENDIF] JAVA_SPEC_VERSION < 17 */
 		SecurityManager manager = System.getSecurityManager();
 		if (manager != null) {
 			manager.checkPermission(CRIU_DUMP_PERMISSION);
 		}
-		/* [ENDIF] JAVA_SPEC_VERSION < 17 */
 
 		setImageDir(imageDir);
 	}
 
+	@SuppressWarnings({ "restriction", "deprecation" })
+	private static void init() {
```

Comment:
I think this should be synchronized as well as it modifies the state that is otherwise protected under lock.  Also, as the only caller is from a synchronized method, documenting that it's synchronized should be nearly free (2nd enter on lock)